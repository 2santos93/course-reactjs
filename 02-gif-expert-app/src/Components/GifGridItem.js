const GifGridItem = ({id, url, title}) => {

    return (
        <div className="card">
            <img src={url} alt={title} className="card-image" />
            <p key={id}>{title}</p>
        </div>
        );

}

export default GifGridItem;