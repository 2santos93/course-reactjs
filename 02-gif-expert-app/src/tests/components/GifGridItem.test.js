import { shallow } from 'enzyme';

import GifGridItem from '../../Components/GifGridItem';
describe('tests for GifGridItem', () => {
    const title = "This is a title";
    const url = "http://localhost";

    const wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should render ok', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have an image and paragraph with their info', () => {

        const p = wrapper.find('p').text();
        const {src, alt, className} = wrapper.find('img').props();

        expect(p).toBe(title);
        expect(src).toBe(url);
        expect(alt).toBe(title);
        expect(className).toBe('card-image');

    });

});
