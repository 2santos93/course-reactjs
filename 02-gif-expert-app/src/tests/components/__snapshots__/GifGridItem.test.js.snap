// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for GifGridItem should render ok 1`] = `
<div
  className="card"
>
  <img
    alt="This is a title"
    className="card-image"
    src="http://localhost"
  />
  <p>
    This is a title
  </p>
</div>
`;
